### PR TITLE
[gas] Fix storage fee on main gas page

### PR DESCRIPTION
**Pull request by gregnazario**
Now, gas fee is still the gas charged, but there is a new field that tells the net gas changes.  It additionally makes the balance changes more user friendly.

Before:
https://explorer.aptoslabs.com/txn/610075380/userTxnOverview?network=mainnet
https://explorer.aptoslabs.com/txn/610075380/balanceChange?network=mainnet
https://explorer.aptoslabs.com/txn/1770359587/userTxnOverview?network=mainnet

After:
...

Resolves: https://github.com/aptos-labs/explorer/issues/778

### CODE REVIEW

File: src/pages/Transactions/TransactionsTable.tsx [hunk 1]
```diff
@@ -191,6 +191,8 @@ function TransactionAmountGasCell({
               <GasFeeValue
                 gasUsed={transaction.gas_used}
                 gasUnitPrice={transaction.gas_unit_price}
+                transactionData={transaction}
+                netGasCost={true}
```
Comment (kaw2k) [inline src/pages/Transactions/TransactionsTable.tsx:195]: nit: you don't need the `={true}`

File: src/pages/Transaction/utils.tsx [hunk 2]
```diff
@@ -227,24 +231,68 @@ export function useTransactionBalanceChanges(txn_version: string) {
...
+        amount: convertAmount(a),
+        type: convertType(a),
         asset: {
           decimals: a.metadata?.decimals,
           symbol: a.metadata?.symbol,
         },
       })) ?? [];
 
+  // Find gas fee and add a storage refund event
+  const gasFeeEvent = data?.fungible_asset_activities.find((a) =>
```
Comment (kaw2k) [inline src/pages/Transaction/utils.tsx:281]: Is StorageRefund an event we are creating? Do we need to account for it in this check?

File: src/pages/Transaction/Tabs/UserTransactionOverviewTab.tsx [hunk 3]
```diff
@@ -151,10 +151,37 @@ export default function UserTransactionOverviewTab({
...
         />
+        <ContentRow
+          title="Storage Refund:"
+          value={
+            <GasFeeValue
+              gasUsed={transactionData.gas_used}
+              gasUnitPrice={transactionData.gas_unit_price}
+              showGasUsed
+              transactionData={transactionData}
+              storageRefund={true}
```
Comment (kaw2k) [inline src/pages/Transaction/Tabs/UserTransactionOverviewTab.tsx:167]: nit: drop `={true}`, same on line 183

File: src/components/IndividualPageContent/ContentValue/GasFeeValue.tsx [hunk 4]
```diff
@@ -2,30 +2,90 @@ import React from "react";
...
+            {" ("}
+            <GasValue gas={gasUsed} />
+            {")"}
+          </span>
+        )}
+      </>
+    );
+  }
+
+  return getGasFee();
```
Comment (kaw2k) [inline src/components/IndividualPageContent/ContentValue/GasFeeValue.tsx:90]: Standard react style is to have your JSX in your return, preferably not nested in a function invocation. If you need a function invocation, using something like `useMemo` would help. I would recommend something like this:  

```tsx
  const feeStatement = transactionData.events?.find(
      (e) => e.type === "0x1::transaction_fee::FeeStatement",
    );

    if (!feeStatement) {
      return (
        <>
        <APTCurrencyValue
          amount={(BigInt(gasUnitPrice) * BigInt(gasUsed)).toString()}
        />
        {showGasUsed === true && (
          <span style={{color: grey[450]}}>
            {" ("}
            <GasValue gas={gasUsed} />
            {")"}
          </span>
        )}
      </>
      );
    }

    const execution_gas = feeStatement.data.execution_gas_units;
    const io_gas = feeStatement.data.io_gas_units;
    const storage_gas = feeStatement.data.storage_fee_octas;
    const storage_refund = feeStatement.data.storage_fee_refund_octas;

    const gross_gas_units =
      BigInt(execution_gas) + BigInt(io_gas) + BigInt(storage_gas);
    const gross_gas_cost = gross_gas_units * BigInt(gasUnitPrice);
    const net_gas_cost = gross_gas_cost - BigInt(storage_refund);

    if (storageRefund) {
      return (
        <>
          <APTCurrencyValue amount={storage_refund.toString()} />
        </>
      );
    }

    if (netGasCost) {
      return (
        <>
          {net_gas_cost < 0 ? "-" : ""}
          <APTCurrencyValue amount={net_gas_cost.toString()} />
        </>
      );
    }

    return (
      <>
        <APTCurrencyValue amount={gross_gas_cost.toString()} />
        {showGasUsed === true && (
          <span style={{color: grey[450]}}>
            {" ("}
            <GasValue gas={gross_gas_units.toString()} />
            {")"}
          </span>
        )}
      </>
    );
```

File: src/pages/Transaction/utils.tsx (hunk 2 above)
Comment (gregnazario) [reply src/pages/Transaction/utils.tsx:281]: It's not a real event, I made one up so the UI looks nice.

Gas fee event is also not a real one, but it's created on the indexer

**Review by kaw2k — APPROVED** [review]

**Pull request merged**